File: planner/app/src/main/java/com/sunilson/firenote/presentation/views/ChecklistView.java
package com.sunilson.firenote.presentation.views;

import android.content.Context;
import android.graphics.Paint;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.CheckBox;
import android.widget.Checkable;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.sunilson.firenote.R;

/**
 * @author LInus Weiss
 */

/**
 * View for a checklist element in a Checklist. Can be checked.
 */
public class ChecklistView extends LinearLayout implements Checkable {

    public View v;
    private TextView textView;
    private CheckBox checkBox;
    private boolean checked = false;
    private LayoutInflater inflater;

    public ChecklistView(Context context) {
        super(context);
        this.inflater = LayoutInflater.from(context);
        v = inflater.inflate(R.layout.checklist_list_layout, this, false);
        textView = (TextView) v.findViewById(R.id.checkList_element_text);
        checkBox = (CheckBox) v.findViewById(R.id.checkList_element_checkBox);
    }

    public TextView getTextView() {
        return textView;
    }

    @Override
    public void setChecked(boolean checked) {
        this.checked = checked;

        if (isChecked()) {
            //Strike through text and show disabled color
            textView.setPaintFlags(textView.getPaintFlags() | Paint.STRIKE_THRU_TEXT_FLAG);
            textView.setTextColor(getResources().getColor(R.color.text_crossed_out));
            checkBox.setChecked(true);
        } else {
            //Display text normally
            textView.setPaintFlags(textView.getPaintFlags() & ~(Paint.STRIKE_THRU_TEXT_FLAG));
            textView.setTextColor(getResources().getColor(R.color.primary_text_color));
            checkBox.setChecked(false);
        }
    }

    @Override
    public boolean isChecked() {
        return checked;
    }

    @Override
    public void toggle() {
    }
}

File: planner/app/src/main/java/com/sunilson/firenote/presentation/shared/views/ChecklistView.kt
package com.sunilson.firenote.presentation.shared.views

import android.content.Context
import android.graphics.Paint
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import com.sunilson.firenote.R
import kotlinx.android.synthetic.main.checklist_list_layout.view.*

class ChecklistView(context: Context) : LinearLayout(context) {

    private val inflater: LayoutInflater = LayoutInflater.from(context)
    val view: View

    init {
        view = inflater.inflate(R.layout.checklist_list_layout, this, false)
    }

    var checked: Boolean = false
        set(value) {
            field = value
            if(value) {
                checkList_element_text.paintFlags = checkList_element_text.paintFlags or Paint.STRIKE_THRU_TEXT_FLAG
                checkList_element_text.setTextColor(ContextCompat.getColor(context, R.color.text_crossed_out))
                checkList_element_checkBox.isChecked = value
            } else {
                checkList_element_text.paintFlags = checkList_element_text.paintFlags and Paint.STRIKE_THRU_TEXT_FLAG.inv()
                checkList_element_text.setTextColor(ContextCompat.getColor(context, R.color.primary_text_color))
                checkList_element_checkBox.isChecked = value
            }
        }
}